smack-core/src/main/java/org/jivesoftware/smack/ConnectionConfiguration.java

```diff
@@ -490,6 +490,18 @@ public Locale getLanguage() {
         return language;
     }
 
+    /**
+     * Returns the xml:lang string of the stream language to use when connecting to the server.
+     *
+     * @return the stream language to use when connecting to the server.
+     */
+    public String getXmlLang() {
+        // TODO: use language.getLanguageTag() once we are on Java1.8 / Android r21+
+        // This will need a workaround for new Locale("").getLanguageTag() returning "und".
+        // Discussion in https://github.com/igniterealtime/Smack/pull/325
```

Comment:
I would prefer an in-code comment summarizing the discussion, but this is also ok.